guillotina/db/transaction.py

```diff
@@ -556,7 +556,13 @@ async def get_children(self, parent, keys):
 
     @profilable
     async def contains(self, oid: str, key: str) -> bool:
-        return await self._manager._storage.has_key(self, oid, key)  # noqa
+        if not await self._manager._storage.has_key(self, oid, key):  # noqa
+            return False
+
+        for obj in self.deleted.values():
+            if obj.id == key:
```

Comment:
I think you also need to make sure the object's parent uid equals `oid` right?

Reply:
right!